components/event-receiver/event-input-adapters/org.wso2.carbon.event.input.adapter.soap/src/main/java/org/wso2/carbon/event/input/adapter/soap/SOAPEventAdapter.java

```diff
@@ -162,26 +162,26 @@ private void registerService(InputEventAdapterListener eventAdaptorListener, Str
 
     private void unregisterService(String serviceName,
                                    AxisConfiguration axisConfiguration) throws AxisFault {
-
         AxisService axisService = axisConfiguration.getService(serviceName);
-
-        if (axisService == null) {
-            throw new AxisFault("There is no service with the name " + serviceName);
-        }
-
-        AxisOperation axisOperation = axisService.getOperation(new QName(SOAPEventAdapterConstants.OPERATION_NAME));
-        if (axisOperation == null) {
-            throw new AxisFault("There is no operation with the name " + SOAPEventAdapterConstants.OPERATION_NAME);
+        try{
+            if (axisService == null) {
+                throw new AxisFault("There is no service with the name " + serviceName);
+            }
+            AxisOperation axisOperation = axisService.getOperation(new QName(SOAPEventAdapterConstants.OPERATION_NAME));
+            if (axisOperation == null) {
+                throw new AxisFault("There is no operation with the name " + SOAPEventAdapterConstants.OPERATION_NAME);
+            }
+            SubscriptionMessageReceiver messageReceiver =
+                    (SubscriptionMessageReceiver) axisOperation.getMessageReceiver();
+            if (messageReceiver == null) {
+                throw new AxisFault("There is no message receiver for operation with name "
+                        + SOAPEventAdapterConstants.OPERATION_NAME);
+            }
+        }finally{
```

Comment:
Thanks @lasanthafdo, Done in pull request - https://github.com/wso2/carbon-analytics-common/pull/109